lib/main.go

```diff
@@ -214,7 +215,29 @@ func (obj *Main) Run() error {
 		return fmt.Errorf("hostname cannot be empty")
 	}
 
+	// Get current user running mgmt
+	user, err := user.Current()
+	if err != nil {
+		return errwrap.Wrapf(err, "can't get current user")
+	}
+
+	// Use systemd SateDirectory if set. If not, use
+	// XDG_CACHE_DIR unless user is root, then use
+	// /var/lib/mgmt
 	var prefix = fmt.Sprintf("/var/lib/%s/", obj.Program) // default prefix
+	stateDir := os.Getenv("STATE_DIRECTORY")
+	xdg := os.Getenv("XDG_CACHE_HOME")
+
+	if xdg == "" {
+		xdg = fmt.Sprintf("%s/.cache/%s", user.HomeDir, obj.Program)
+	}
+
+	if stateDir != "" {
+		prefix = stateDir
+	} else if user.Uid != "0" {
+		prefix = fmt.Sprintf("%s", xdg)
```

Comment:
unnecessary sprintf

Reply:
Sorry, I thought I tested it without that and it didn't work. I ran it again and it did. My bad.